app/workers/database_index_worker.rb

```diff
@@ -0,0 +1,65 @@
+class DatabaseIndexWorker
+  include Sidekiq::Worker
+  sidekiq_options queue: :async_database_migration, retry: 3
+
+  def perform(action = 'create')
+    case action
+    when 'create'
+      create_indexes
+    when 'drop'
+      drop_indexes
+    end
+  end
+
+  private
+
+  def create_indexes
+    setup_environment
+    create_composite_index
+    create_trigram_index
+    reset_timeout
+  end
+
+  def drop_indexes
+    setup_long_timeout
+    drop_new_indexes
+    reset_timeout
+  end
+
+  def setup_environment
+    # First create the trigram extension (needs to be in its own transaction)
+    ActiveRecord::Base.connection.execute 'CREATE EXTENSION IF NOT EXISTS pg_trgm;'
+    setup_long_timeout
+  end
+
+  def setup_long_timeout
+    # Set a very long statement timeout (24 hours) for this connection only
+    ActiveRecord::Base.connection.execute "SET statement_timeout = '86400000';" # 24 hours in milliseconds
```

Comment:
While the connection might not timeout, Sidekiq might, can you confirm the default timeout for Sidekiq? Ideally it should be greater than the statement timeout, allowing it to run any cleanup tasks.